ci/azure/macos.yml

```diff
@@ -0,0 +1,47 @@
+# adapted from pandas' version
+
+parameters:
+  name: ''
+  vmImage: ''
+
+jobs:
+- job: ${{ parameters.name }}
+  pool:
+    vmImage: ${{ parameters.vmImage }}
+  strategy:
+    maxParallel: 11
+    matrix:
+      py35_np_113:
+        ENV_FILE: ci/azure-macos-35.yml
+        CONDA_PY: "35"
+        CONDA_ENV: sm2
+        TEST_ARGS: "--skip-slow"
+
+  steps:
+    - script: |
+        if [ "$(uname)" == "Linux" ]; then sudo apt-get install -y libc6-dev-i386; fi
+        echo "Installing Miniconda"
+        ci/incremental/install_miniconda.sh
+        export PATH=$HOME/miniconda3/bin:$PATH
+        echo "Setting up Conda environment"
+        ci/incremental/setup_conda_environment.sh
+      displayName: 'Before Install'
+    - script: |
+        export PATH=$HOME/miniconda3/bin:$PATH
+        ci/incremental/build.sh
+      displayName: 'Build'
+    - script: |
+        export PATH=$HOME/miniconda3/bin:$PATH
+        ci/script_single.sh
+        echo $?
+        ci/script_multi.sh
+        echo $?
```

Comment:
> The publish test results task will not fail the build based on failed test results alone. You will need to have the test script return a non-zero exit code to fail the build.

@chrisrpatterson both here and two lines above I'm seeing a `1`, as expected.  Is there somewhere else where a non-zero exit code needs to be returned?

Reply:
Can you try adding set -o pipefail as the first line of your script?  I suspect echo returns non-zero and it is the last line.

Reply:
@chrisrpatterson thanks for the suggestion.  Added that on line 34 above, no clear difference.